Fail to enable `http` feature of `jsonrpc-core-client`

I enabled `http` feature of `jsonrpc-core-client` and I got this issue:

```
error[E0599]: no function or associated item named `new` found for struct `Client` in the current scope
  --> /home/yvan/.cargo/registry/src/github.com-1ecc6299db9ec823/jsonrpc-client-transports-18.0.0/src/transports/http.rs:32:23
   |
32 |     let client = Client::new();
   |                          ^^^ function or associated item not found in `Client<_, _>`
```

This occurs in an empty `fn main() {}` program, just by adding `jsonrpc-core-client = { version = "18.0.0", features = ["http"] }` line in `Cargo.toml` (you can found the full demo program here: https://github.com/yvan-sraka/jsonrpc-client-http-demo).

Due to the lack of documentation, I couldn’t determine if this behavior is normal (and I just didn’t find out yet how to use the feature) or if it shouldn’t fail this way?

I was using Rust nightly and I test that the issue didn’t occurs with previous version `17.0.0` of `jsonrpc-core-client` crate …

--- a/.editorconfig
+++ b/.editorconfig
@@ -9,7 +9,7 @@ trim_trailing_whitespace=true
 max_line_length=120
 insert_final_newline=true
 
-[{.travis.yml,appveyor.yml}]
+[{.travis.yml,appveyor.yml,.gitlab-ci.yml}]
 indent_style=space
 indent_size=2
 tab_width=8

--- a/.gitlab-ci.yml
+++ b/.gitlab-ci.yml
@@ -10,6 +10,7 @@ variables:                         &default-vars
   script:
     - cargo build --all
     - cargo test --all
+    - cargo build --manifest-path ./core-client/Cargo.toml --no-default-features --features http
 
 .only:                             &only
   only:

--- a/core-client/transports/Cargo.toml
+++ b/core-client/transports/Cargo.toml
@@ -44,7 +44,7 @@ serde = { version = "1.0", features = ["derive"] }
 serde_json = "1.0"
 url = "1.7"
 
-hyper = { version = "0.14", features = ["client", "http1"], optional = true }
+hyper = { version = "0.14", features = ["client", "http1", "tcp"], optional = true }
 hyper-tls = { version = "0.5", optional = true }
 jsonrpc-server-utils = { version = "18.0.0", path = "../../server-utils", optional = true }
 parity-tokio-ipc = { version = "0.9", optional = true }

